src/main/java/xbdd/webapp/resource/feature/Report.java

```diff
@@ -364,19 +362,28 @@ public DBObject putReport(@BeanParam final Coordinates coordinates, final DBObje
 			this.log.trace("Adding feature:" + JSON.serialize(feature));
 			features.save(feature);
 		}
-		final DBCursor cursor = features.find(coordinates.getReportCoordinatesQueryObject()); // get new co-ordinates to exclude the "version"
-																						// field
-		final List<DBObject> returns = new ArrayList<DBObject>();
+		final DBCursor cursor = features.find(coordinates.getReportCoordinatesQueryObject()); // get new co-ordinates to exclude the
+																								// "version"
+		updateStatsDocument(bdd, coordinates, cursor);
+		return extractList(cursor);
+	}
+
+	/**
+	 * Collects the items available to the cursor into a list.
+	 *
+	 * @param cursor The cursor information to use. Iteration is performed on a copy of this cursor so there are no side effects.
+	 * @return list of all items available to the cursor
+	 */
+	private static BasicDBList extractList(final DBCursor cursor) {
+		final DBCursor cursorCopy = cursor.copy(); // no side effects please
```

Comment:
The cursor.next() method advances the cursor to the next object, so if we don't copy it, this method "uses up" the cursor. I chose to copy the parameter cursor so that it was not changed as a side-effect of the method. But if you're happy with side effects I'll document it and remove the copy.